### PR TITLE
[7X] Fix FNullRejecting() error while trying to process zero-placed predicate

### PR DESCRIPTION
**Pull request by InnerLife0**
When trying to do NULL rejection over zero-placed predicate, e.g. any strict function with zero arguments, we may pass NULL children array from EberEvaluate() to underlying Eber() functions. This may lead to errors, such as assertion error at EberNullOnAnyNullChild() function which used by many subclasses. As underlying Eber() functions doesn't perform any NULL check, we need to pass an empty array to them in case we working with zero-placed predicate.
Another option was to return EberAny from EberEvaluate() in case we dealing with zero arity, but there is underlying CScalarConst::Eber() function which should be called and which do it's stuff ignoring incoming argument at all. Here I decided to keep current solution as it's clean.

A little highlighting of function folding. If you'll remove topmost select from any of new tests, you'll not get an error because strict function will be folded, and thus, became Const. Starting from 4cf47434a35df73dd456b6648c5f40ee377b5845 such functions under topmost select not folded, because such select has no parse->rtable. At the other hand, Postgres planner, on which ORCA fell back before current patch, can do folding because of separate calls to eval_const_expressions() from subplans.

### CODE REVIEW

**Review by chrishajas — APPROVED** [review]
Thanks for this PR and test case! I verified that this also causes a crash in retail builds before this patch.

File: src/test/regress/sql/gporca.sql [hunk 1]
```diff
@@ -3435,6 +3435,39 @@ DROP TABLE IF EXISTS dist_tab_a;
...
+$$ LANGUAGE plpgsql STABLE STRICT;
+
+EXPLAIN (COSTS OFF) SELECT (
+    SELECT count(*) cnt
+    FROM join_null_rej1 t1
+    LEFT JOIN join_null_rej2 t2 ON t1.i = t2.i
+    WHERE t2.i < join_null_rej_func()
+);
+-- Optional, but let's check we get same result for both, folded and
+-- not folded now() frunction.
```
Comment (bhuvnesh2703) [inline src/test/regress/sql/gporca.sql:3463]: - typo `frunction`
- `also folded and not folded now()` looks like this statement need to be corrected to represent the join_null_rej_func func.

**Review by bhuvnesh2703 — APPROVED** [review]
Minor Comments. but LGTM

File: src/test/regress/sql/gporca.sql (hunk 1 above)
Comment (InnerLife0) [reply src/test/regress/sql/gporca.sql]: Thanks! Fixed.

**Review by khannaekta — APPROVED** [review]
LGTM!

**Comment by bhuvnesh2703** [comment]
Thank you @InnerLife0 for the contribution.

**Comment by InnerLife0** [comment]
> Thank you @InnerLife0 for the contribution. Please squash the commit, i will merge after it.

Thanks. Squashed for both, 7X and 6X.

**Pull request merged**